src/openrct2/world/TileInspector.cpp

```diff
@@ -30,1162 +30,1104 @@
+                    // to do is to invalidate the window.
+                    break;
+                }
+            }
 
-        // Update the tile inspector's list for everyone who has the tile selected
-        rct_window* const tileInspectorWindow = window_find_by_class(WC_TILE_INSPECTOR);
-        if (tileInspectorWindow != nullptr && loc == windowTileInspectorTile.ToCoordsXY())
-        {
-            windowTileInspectorElementCount++;
+            map_invalidate_tile_full(loc);
 
-            // Keep other elements (that are not being hidden) selected
-            if (windowTileInspectorSelectedIndex > elementIndex)
+            if (auto* inspector = GetTileInspectorWithPos(loc); inspector != nullptr)
             {
-                windowTileInspectorSelectedIndex++;
+                windowTileInspectorElementCount++;
+                if (windowTileInspectorSelectedIndex > elementIndex)
+                {
+                    windowTileInspectorSelectedIndex++;
+                }
+                inspector->Invalidate();
             }
-
-            tileInspectorWindow->Invalidate();
         }
-    }
 
-    // Nothing went wrong
-    return std::make_unique<GameActions::Result>();
-}
+        // Nothing went wrong
+        return std::make_unique<GameActions::Result>();
+    }
 
-static int32_t numLargeScenerySequences(const CoordsXY& loc, const LargeSceneryElement* const largeScenery)
-{
-    const rct_scenery_entry* const largeEntry = largeScenery->GetEntry();
-    const auto* const tiles = largeEntry->large_scenery.tiles;
-    const auto direction = largeScenery->GetDirection();
-
-    const auto rotatedFirstTile = CoordsXYZ{
-        CoordsXY{ tiles[largeScenery->GetSequenceIndex()].x_offset, tiles[largeScenery->GetSequenceIndex()].y_offset }.Rotate(
-            direction),
-        tiles[largeScenery->GetSequenceIndex()].z_offset
-    };
-
-    const auto firstTile = CoordsXYZ{ loc, largeScenery->GetBaseZ() } - rotatedFirstTile;
-    auto numFoundElements = 0;
-    for (int32_t i = 0; tiles[i].x_offset != -1; i++)
+    static int32_t numLargeScenerySequences(const CoordsXY& loc, const LargeSceneryElement* const largeScenery)
     {
-        const auto rotatedCurrentTile = CoordsXYZ{ CoordsXY{ tiles[i].x_offset, tiles[i].y_offset }.Rotate(direction),
-                                                   tiles[i].z_offset };
+        const rct_scenery_entry* const largeEntry = largeScenery->GetEntry();
+        const auto* const tiles = largeEntry->large_scenery.tiles;
+        const auto direction = largeScenery->GetDirection();
+
+        const auto rotatedFirstTile = CoordsXYZ{ CoordsXY{ tiles[largeScenery->GetSequenceIndex()].x_offset,
+                                                           tiles[largeScenery->GetSequenceIndex()].y_offset }
+                                                     .Rotate(direction),
+                                                 tiles[largeScenery->GetSequenceIndex()].z_offset };
```

Comment:
A trailing comma will likely make this readable:

```suggestion
        const auto rotatedFirstTile = CoordsXYZ{
            CoordsXY{ tiles[largeScenery->GetSequenceIndex()].x_offset, tiles[largeScenery->GetSequenceIndex()].y_offset }
                .Rotate(direction),
            tiles[largeScenery->GetSequenceIndex()].z_offset,
        };
```

Reply:
I can probably do a little better by having sequence index being a separate variable also.

Reply:
Still pretty hefty but a bit better now.